setup.py

```diff
@@ -1,11 +1,31 @@
 from setuptools import setup, Extension
 import os, platform, sys, re
 
-# look for environment variable that specifies path to SCIP Opt lib and headers
+# look for environment variable that specifies path to SCIP
 scipoptdir = os.environ.get('SCIPOPTDIR', '').strip('"')
-includedir = os.path.abspath(os.path.join(scipoptdir, 'include'))
+
+# now check whether SCIP is installed on the system or whether a source code directory has been specified
+if os.path.exists(os.path.join(scipoptdir, 'include')):
+    includedir = os.path.abspath(os.path.join(scipoptdir, 'include'))
+else:
+    if os.path.exists(os.path.join(scipoptdir, 'src')):
+        includedir = os.path.abspath(os.path.join(scipoptdir, 'src'))
+    else:
+        print('Neither directory \'src\' nor \'include\' exists in the path <%s>' % scipoptdir)
+        quit(1)
+
+# determine library
 libdir = os.path.abspath(os.path.join(scipoptdir, 'lib'))
 libname = 'scip'
+if (not os.path.exists(os.path.join(libdir, 'libscip.so'))) and (not os.path.exists('/usr/lib/libscip.so')):
```

Comment:
This is Linux-only and makes the setup routine very convoluted. On macOS the libs have the suffix `dylib` and on Windows there are both `lib` and `dll`.

Reply:
I could turn the test around and only check for scipoptsolver. If it is there, then one can use it. Otherwise one falls back to the default.

Another option would be to use  ``find_library``  from the ``ctypes.util`` package. This should be a portable way to check for a library, e.g., ``find_library('m')`` would find ``libm.so`` on Linux.

Reply:
I don't want to use `find_library()` because we might then pick a system library over a user specified one.

Reply:
So how is the SCIP version chosen? If ``SCIPOPTDIR`` is set (and it exists), then this should be used, right? I would only use ``find_library`` if the SCIP directory cannot be found.

Reply:
I have changed the behavior - see the other comments.